### DISCUSSION TITLE
Two dependent input dropdowns

### DISCUSSION THREAD

**Discussion by leostone68** [elsa-workflows/elsa-core#5641] [Q&A] [answered]
dio design environment, I would like to manage an activity, defined as having two dropdown inputs. By selecting a value from the first dropdown I should read the value and populate the second dropdown by calling an API that extracts a list of values. Thank you

**Comment by glime-ai** [comment, marked as answer]
To implement a cascading dropdown in Elsa Workflow within an activity, you would typically need to use JavaScript to handle the change event of the first dropdown and then make an API call to retrieve the values for the second dropdown based on the selected value of the first dropdown.

Here's a general approach to achieve this:

1. **Create the First Dropdown**: Define the first dropdown in your activity with static values or populate it dynamically using an OptionsProvider.

2. **Handle the Change Event**: Use JavaScript to listen for changes on the first dropdown. When a change occurs, capture the selected value.

3. **Make an API Call**: Use the captured value to make an API call to the backend service that will return the list of values for the second dropdown.

4. **Populate the Second Dropdown**: Once you receive the response from the API, populate the second dropdown with the new set of options.

5. **Update the Workflow Definition**: Ensure that your workflow definition is updated to reflect the dynamic nature of the second dropdown.

Unfortunately, the provided context does not include specific examples of implementing such JavaScript within the Elsa Workflow designer environment. However, based on the context from the GitHub issues and Stack Overflow questions, it seems that dynamic population of dropdowns is a feature that has been discussed and is in demand within the Elsa community.

If you are using Elsa 3, you might need to wait for the feature to be ported as per the GitHub issue #4491, or you could attempt to implement a custom solution by extending the Elsa activities and using JavaScript to interact with the API and update the dropdown options dynamically.

For a more detailed and specific solution, you would need to refer to the Elsa Workflow documentation or seek assistance from the Elsa community, as the current context does not provide a direct example of implementing cascading dropdowns in Elsa Workflow.

If you have access to the Elsa Workflow documentation or forums, I would recommend searching there for more detailed guidance or examples. Additionally, you may want to follow the GitHub issues related to dynamic option population for updates on when this feature might be available in Elsa 3.

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Reply by leostone68** [reply]
Thanks for the reply. I suspected there wasn't this feature.
